docs/content/guides/dagster/managing-ml.mdx

```diff
@@ -0,0 +1,266 @@
+        yield Output(reg, metadata={"model_accuracy": new_model_accuracy})
+```
+
+We can also monitor the score or other information of a model through alerts, such as email or Slack. In the example below, we set up a Slack message to alert the score of ml_model upon success. The message can be customized to include information such as which features were most influential in model training, the volume of data that the model was trained on or in this case, and the score of the model.
+
+```python file=/guides/dagster/managing_ml/managing_ml_code.py startafter=success_slack_start  endbefore=success_slack_end
+from dagster import asset_sensor, RunRequest, asset
+from sklearn import linear_model
+
+
+@asset
+def ml_model(training_data, test_data, model_accuracy):
+    reg = linear_model.LinearRegression()
+    reg.fit(training_data)
+    new_model_accuracy = reg.score(test_data)
+    if new_model_accuracy > model_accuracy:
+        yield Output(reg, metadata={"model_accuracy": new_model_accuracy})
+```
+
+We can also set up a sensor that triggers if an asset fails to materialize. Alerts can be customized and sent through e-mail or natively through Slack. In this example, a Slack message is sent anytime the `ml_job` fails.
+
+```python file=/guides/dagster/managing_ml/managing_ml_code.py startafter=fail_slack_start   endbefore=fail_slack_end
+import os
+from dagster import make_slack_on_run_failure_sensor, define_asset_job
+
+ml_job = define_asset_job("ml_training_job", selection=[ml_model])
+
+slack_on_run_failure = make_slack_on_run_failure_sensor(
+    channel="#ml_monitor_channel",
+    slack_token=os.getenv("MY_SLACK_TOKEN"),
+    monitored_jobs=(["ml_job"]),
+)
+```
+
+## Enhancing the Dagster UI with metadata
+
+Understanding the performance of your ML model is critical to both the model development process and production. Metadata can significantly enhance the usability of the Dagster UI to show what’s going on in a specific asset. Using metadata in Dagster is flexible, can be used for tracking evaluation metrics, and viewing the training accuracy progress over training iterations as a graph.
+
+One of the easiest ways to utilize Dagster’s metadata is by using a dictionary to track different metrics that are relevant for an ML model.
+
+Another way is to store relevant data for a single training iteration as a graph that you can view directly from the Dagster UI. In this example, we’ll define a function that uses data produced by a machine learning model to plot an evaluation metric as the model goes through the training process and render that in the Dagster UI.
+
+Dagster’s [`MetadataValue` types](/\_apidocs/ops#event-metadata>) enables types such as tables, URLs, notebooks and Markdown, etc. In the following example, the Markdown metadata type is used to generate plots. Each plot will show a specific evaluation metric’s performance throughout each training iteration also known as an epoch during the training cycle.
```

Comment:
```suggestion
Dagster’s <PyObject object="MetadataValue" /> types enable types such as tables, URLs, notebooks, Markdown, etc. In the following example, the Markdown metadata type is used to generate plots. Each plot will show a specific evaluation metric’s performance throughout each training iteration also known as an epoch during the training cycle.
```

- There's a `PyObject` we can use for `MetadataValue` - check that it points to the spot you wanted, but I think this is correct
- Fixes subject-verb agreement
- Add a comma to series